### ISSUE TITLE
Unable to Invoke Docker Action - OpenWhisk

### ISSUE THREAD

**Issue by kameshsampath** [spring-cloud/spring-cloud-function#141]
I tried to build a simple Spring Cloud Function which am deploying to OpenWhisk running locally,

Using : Spring Cloud Functions 1.0.0.BUILD-SNAPSHOT

App Class:
----------
```
package functions;
... (imports)
@SpringBootApplication
public class Application {

    @Bean
    public Function<Greetings, Greetings> greeter() {
        return v -> new Greetings(v.getName());
    }
... (psvm)
}
```

Greetings is just POJO with name and message members

function.properties
-------------------
```
dependencies.function: com.example\:hello-pof\:0.0.1-SNAPSHOT
```

After build the docker image as https://github.com/spring-cloud/spring-cloud-function/tree/master/spring-cloud-function-adapters/spring-cloud-function-adapter-openwhisk#README.md section  "Use the following Dockerfile" 

Once the action is created, I tried invoking the action using the command 

`wsk action invoke greeter -param payload '{"name": "tom"}'`, resulting in 

```
2018-02-07T13:09:36.975505203Z stdout: 2018-02-07 13:09:36.970 ERROR 1 --- [nio-8080-exec-2] o.a.c.c.C.[.[.[/].[dispatcherServlet]    : Servlet.service() for servlet [dispatcherServlet] in context with path [] threw exception [Request processing failed; nested exception is java.lang.ClassCastException: java.util.LinkedHashMap cannot be cast to reactor.core.publisher.Flux] with root cause",
    "2018-02-07T13:09:36.975541701Z stdout: ",
    "2018-02-07T13:09:36.975546325Z stdout: java.lang.ClassCastException: java.util.LinkedHashMap cannot be cast to reactor.core.publisher.Flux",
    "2018-02-07T13:09:36.975550156Z stdout: at org.springframework.cloud.function.core.FluxFunction.apply(FluxFunction.java:32) ~[spring-cloud-function-core-1.0.0.BUILD-SNAPSHOT.jar:1.0.0.BUILD-SNAPSHOT]",
    "2018-02-07T13:09:36.975553666Z stdout: at org.springframework.cloud.function.core.FluxFunction.lambda$apply$0(FluxFunction.java:42) ~[spring-cloud-function-core-1.0.0.BUILD-SNAPSHOT.jar:1.0.0.BUILD-SNAPSHOT]",
```

The complete openwhisk activation result log is available here --> https://gist.github.com/kameshsampath/7c53103b4c42c2ffcd29ff1d5b8f7636

**Comment by kameshsampath** [comment]
@dsyer - think I have fixed the issue in my branch https://github.com/kameshsampath/spring-cloud-function/tree/ow-issue-141, please let me know if you are OK to me sending PR, I can then sanitise and send a PR your way 

I also wish to sanitise the class names inline with AWS/Azure  and add some documentation especially for OpenWhisk

**Comment by dsyer** [comment]
Sounds good. I think it would be better to inject the `ObjectMapper` (Spring Boot will provide one if it is on the classpath), but otherwise I didn't see any issues.

**Comment by kameshsampath** [comment]
Thanks @dsyer; I thought the autowiring of it inline with other adapters will add as part of my PR, will clean my experimental branch and send you PR by EoD my time .How about refactor classnames inline with other Adapters e.g FunctionIntializer can be like OpenWhiskFunctionIntializer??

**Comment by dsyer** [comment]
Fine by me

**Comment by kameshsampath** [comment]
@dsyer another question i have right now is, if the user does not pass the `payload` parameter then the app throws a NPE, i feel we should still return a result with empty string

**Comment by dsyer** [comment]
I don't know what a sensible default is. NPE not. Happy to go with whatever you decide for now.

**Comment by dsyer** [comment]
Closed with #142
